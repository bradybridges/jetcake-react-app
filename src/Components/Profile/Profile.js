import React, { Component } from 'react';
import ProfileInformation from '../ProfileInformation/ProfileInformation';
import EditProfileForm from '../EditProfileForm/EditProfileForm';
const editImg = require('../../Assets/edit.png');

export default class Profile extends Component{
  state = {
    showEditProfile: false,
  }

  toggleShowEditProfile = () => {
    const updatedState = !this.state.showEditProfile;
    this.setState({ showEditProfile: updatedState });
  }

  render() {
    const {
      email, 
      address, 
      phone, 
      dob, 
      securityOne, 
      securityTwo, 
      securityThree, 
      photoURL, 
      handleUpdateProfile 
    } = this.props;

    return (
      <section id="profile-container">
        <img id="profile-img" src={photoURL} alt="profile" />
        <div 
          id="edit-btn-container" 
          style={{backgroundColor: 'black'}}
          onClick={() => this.setState({ showEditProfile: true })}
        >
          <img src={editImg} alt="edit" />
        </div>
        {this.state.showEditProfile && <EditProfileForm 
          email={email}
          address={address}
          phone={phone}
          dob={dob}
          securityOne={securityOne}
          securityTwo={securityTwo}
          securityThree={securityThree}
          toggleShowEditProfile={this.toggleShowEditProfile}
          handleUpdateProfile={handleUpdateProfile}
        />}
        { !this.state.showEditProfile && <ProfileInformation 
          email={email}
          address={address}
          phone={phone}
          dob={dob}
          securityOne={securityOne}
          securityTwo={securityTwo}
          securityThree={securityThree}
        />
        }
      </section>
    );
  }
}

