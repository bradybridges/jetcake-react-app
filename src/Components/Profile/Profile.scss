#profile-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  color: white;
  min-height: 80vh;
}

#profile-header {
  font-size: 2em;
  margin: 10px 0;
}

#profile-img {
  border-radius: 50%;
  width: 200px;
  height: 200px;
  margin: 20px 0;
}

#edit-btn-container {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: transform .2s;
  position: absolute;
  right: 3%;
  top: 16vh;
  border: solid white 1px;
  background-color: rgba(0, 0, 0, 0.315);
  img {
    width: 25px;
    height: 25px;
  }
  &:hover {
    cursor: pointer;
    transform: scale(1.1);
  }
}

.back-link {
  width: 25%;
  color: whiite;
  color: white;
  text-decoration: none;
  font-weight: bold;
  align-self: flex-start;
  transition: transform .2s;
  padding: 20px;
  &:hover {
    transform: scale(1.1);
  }
}