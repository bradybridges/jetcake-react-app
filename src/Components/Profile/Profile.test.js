import React from 'react';
import { shallow } from 'enzyme';
import Profile from './Profile';

describe('Profile', () => {
  let wrapper;
  const mockHandleUpdateProfile = jest.fn();
  it('should match the snapshot when edit profile is false', () => {
    wrapper = shallow(<Profile
      email="test@test.com"
      address="123 test blvd"
      phone="1112223333"
      dob="1990-10-10"
      securityOne="Roger"
      securityTwo="Fido"
      securityThree="Minneapolis"
      photoURL="https://mockHost.com/userID/profile.jpg"
      handleUpdateProfile={mockHandleUpdateProfile}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when edit profile is true', () => {
    const editTrueWrapper = shallow(<Profile
      email="test@test.com"
      address="123 test blvd"
      phone="1112223333"
      dob="1990-10-10"
      securityOne="Roger"
      securityTwo="Fido"
      securityThree="Minneapolis"
      photoURL="https://mockHost.com/userID/profile.jpg"
      handleUpdateProfile={mockHandleUpdateProfile}
    />);
    editTrueWrapper.setState({ showEditProfile: true });
    expect(editTrueWrapper).toMatchSnapshot();
  });

  it('toggleShowEditProfile should toggle showEditProfile in state', () => {
    expect(wrapper.state('showEditProfile')).toEqual(false);
    wrapper.instance().toggleShowEditProfile();
    expect(wrapper.state('showEditProfile')).toEqual(true);
  });
});
