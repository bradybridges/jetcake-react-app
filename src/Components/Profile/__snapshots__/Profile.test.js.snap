// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match the snapshot when edit profile is false 1`] = `
<section
  id="profile-container"
>
  <h3>
    Profile
  </h3>
  <img
    alt="profile"
    id="profile-img"
    src="https://mockHost.com/userID/profile.jpg"
  />
  <div
    id="edit-btn-container"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    <img
      alt="edit"
      src="edit.png"
    />
  </div>
  <ProfileInformation
    address="123 test blvd"
    dob="1990-10-10"
    email="test@test.com"
    phone="1112223333"
    securityOne="Roger"
    securityThree="Minneapolis"
    securityTwo="Fido"
  />
</section>
`;

exports[`Profile should match the snapshot when edit profile is true 1`] = `
<section
  id="profile-container"
>
  <h3>
    Profile
  </h3>
  <img
    alt="profile"
    id="profile-img"
    src="https://mockHost.com/userID/profile.jpg"
  />
  <div
    id="edit-btn-container"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    <img
      alt="edit"
      src="edit.png"
    />
  </div>
  <EditProfileForm
    address="123 test blvd"
    dob="1990-10-10"
    email="test@test.com"
    handleUpdateProfile={[MockFunction]}
    phone="1112223333"
    securityOne="Roger"
    securityThree="Minneapolis"
    securityTwo="Fido"
    toggleShowEditProfile={[Function]}
  />
</section>
`;
